### PR TITLE
Support nahmii compiler

### PR DESCRIPTION
**Pull request by HDauven**
This pull request contains the following changes:
- Support for the `remix-nahmii-compiler-plugin`.
- Custom context menu for the Nahmii compiler.
- Handling of compilation results for the Nahmii compiler.

### CODE REVIEW

**Comment by HDauven** [comment]
This PR is dependent on the merging of the [following PR](https://github.com/ethereum/remix-plugins-directory/pull/273) in the `remix-plugins-directory`.

**Comment by yann300** [comment]
thanks for the PR, could you rebase it please?

**Comment by ryestew** [comment]
@HDauven thanks for the PR, could you rebase it please?

**Comment by HDauven** [comment]
Hey @yann300 @ryestew! Sorry for the delay and thank you for the heads up.

I had to do a couple of changes due to recent changes in the project structure. (Like the removal of the `dropdownlogic` runTab model) Everything should be working as expected now.

File: apps/remix-ide/src/remixAppManager.js [hunk 1]
```diff
@@ -149,6 +149,16 @@ export class RemixAppManager extends PluginManager {
       pattern: [],
       sticky: true
     })
+    await this.call('filePanel', 'registerContextMenuItem', {
```
Comment (yann300) [inline apps/remix-ide/src/remixAppManager.js:152]: @HDauven note that for this to work, the plugin side should implement the `compileCustomAction` function, is this the case?
cc @bunsenstraat

Comment (yann300) [reply apps/remix-ide/src/remixAppManager.js:152]: We can merge the PR now, but `compileCustomAction` must be added in the plugin codebase.

**Review by yann300 — APPROVED** [review]

File: apps/remix-ide/src/remixAppManager.js (hunk 1 above)
Comment (yann300) [reply apps/remix-ide/src/remixAppManager.js:152]: please reach out in our channel https://gitter.im/ethereum/remix-dev if you want more information about that.

Comment (HDauven) [reply apps/remix-ide/src/remixAppManager.js:152]: Hey @yann300! 

The plugin actually does implement this. There is a hook that extends the PluginClient provided by Remix. This RemixClient adds both a setCompilerCallback callback function, compileCustomAction function and a compile function.

Comment (HDauven) [reply apps/remix-ide/src/remixAppManager.js:152]: Note that I've tested the above mentioned functionality extensively, both regular compilations and compilations through the context menu, both locally and by plugging in the locally ran plugin into the hosted remix environment.

**Pull request merged**